scripts/version_maker.sh

```diff
@@ -18,63 +22,126 @@ function yaml() {
     }'
 }
 
-# find files ended with chartserviceversion.yaml 
-FIND_CMD=`find ./charts -type f -name "*basetemplate.yaml"`
+# This function checks whether the kind is CSV or CE
+checker(){
+    if [[ $1 == "ChaosExperiment" ]]; then
+        echo $metadata_version
+    elif [[ $1 == "ChartServiceVersion" ]]; then
+        echo $spec_version 
+    fi
+}
 
-# add all the files to an array called files
-files=$(echo $FIND_CMD | tr " " "\n")
+# This function takes the old version from the last commit 
+# and increment the existing version by one unit.
+versionInc(){
+    echo "version inc"
+    file=$1
+    
+    eval $(yaml_parser $file)
+    if [[ $? == 0 ]]; then
+        existing_version=`checker $kind`
+        echo "Existing version: $existing_version"
 
-# looping from all the files
-for file in $files
-do
-    # get the latest version from the template file
-    eval $(yaml $file)
+        # stores the last pushed committed file as temp.yaml in the root directory 
+        # and will be deleted after the job
+        temp_file=`git show $last_commit_hash:$1 >> temp.yaml`
+
+        eval $(yaml_parser './temp.yaml')
+        if [[ $? == 0 ]]; then
+
+            oldversion=`checker $kind`
+            echo "Oldversion : $oldversion"
+
+            # storing version to an array 
+            versions=( ${oldversion//./ } )
+            ((versions[2]++)) # Increment the patch version by one unit
+            newversion="${versions[0]}.${versions[1]}.${versions[2]}"
+
+            # This python script checks and validate the version
+            sudo python3 scripts/validate_version.py $oldversion $newversion
+        
+            if [[ $? == 0 ]]; then
+                `sed -i "s/$existing_version/$newversion/" $file` &&
+                echo "$file's version updated from $oldversion to $newversion"
+            fi
+        fi
+        # deleting the temporary file 
+        rm './temp.yaml'
+    fi
+}
+
+# This function takes the latest from the base-template'
+# and increment the existing version in the csv/experiment file
+# by the input version mention in the basetemplate by the user.
+versionMod(){
```

Comment:
Can we try to avoid this (taking majpr and minor version on basetemplate artiifact from the committer  and setting it back to {{ version }} ? 

Instead have a function to update minor/major versions only upon github releases (tags)

To facilitate this, we can have chart versions following release versions of openebs (patch can keep getting incremented until we do the next release)